### ISSUE TITLE
More precise types for MandrillHeaders and MandrillVars

### ISSUE THREAD

**Issue by basvandijk** [adinapoli/mandrill#24]
[MandrillHeaders](http://hackage.haskell.org/package/mandrill-0.4.1.0/docs/Network-API-Mandrill-Types.html#t:MandrillHeaders) is a synonym for [Value](https://hackage.haskell.org/package/aeson-0.10.0.0/docs/Data-Aeson.html#t:Value). However according to the [Mandrill API docs](https://mandrillapp.com/api/docs/messages.JSON.html) it seems to only allow a JSON [Object](https://hackage.haskell.org/package/aeson-0.10.0.0/docs/Data-Aeson.html#t:Object). Do you also think it's better to switch it to `Object`.

Similarly, [MandrillVars](http://hackage.haskell.org/package/mandrill-0.4.1.0/docs/Network-API-Mandrill-Types.html#t:MandrillVars) is also a synonym for `Value`. However the docs specify it should be an array of name&content objects:

```haskell
type MandrillVars = [MandrillVar]

data MandrillVar = MandrillVar
  { _mv_name    :: !Text
  , _mv_content :: !Value
  }
```

Do you think this makes sense?

**Comment by adinapoli** [comment]
I think it makes total sense. To be really honest with you, for the initial release I want with the path of least resistance (ehm, read lazyness) and simply represented them as value.

Thanks for the spot on observation!

**Comment by basvandijk** [comment]
> I think it makes total sense.

Cool, then I will prepare a PR tomorrow.

> Thanks for the spot on observation!

I actually wrote my own mandrill library at Better. However you released yours just before I would :-)

**Comment by adinapoli** [comment]
Hehe, a man has to do what it can to carve a small niche in the Haskell library maintainers, with so many high quality ones around :wink:

**Comment by basvandijk** [comment]
:-)

**Comment by basvandijk** [comment]
Should be fixed by #25.
